helix-view/src/editor.rs

```diff
@@ -271,6 +271,51 @@ impl Editor {
         Ok(())
     }
 
+    /// Refreshes the language server for a given document
+    pub fn refresh_language_server(&mut self, doc_id: DocumentId) -> Option<()> {
+        // don' know why but it feels good using this pattern :)
+        let Editor {
+            documents,
+            language_servers,
+            theme,
+            syn_loader,
+            ..
+        } = self;
```

Comment:
I'd prefer not to do this because it doesn't simplify the code in any way and it makes it less obvious which values are coming from `self` and which are local variables.